### PR TITLE
[Feature&Fix] Add sympy_to_func

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
<!-- Demo: https://github.com/PaddlePaddle/PaddleScience/pull/96 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
New features
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
APIs 
### Describe
<!-- Describe what this PR does -->
主要修改点：
1. **初步**添加对sympy表达式的支持（设计方案：[sympy_to_func](https://github.com/PaddlePaddle/community/blob/master/pfcc/paddle-code-reading/sympy_to_func/sympy_to_func.md)）
2. 添加在 `LinearElasticity` 和 `NavierStokes` 上的单测，测试结果如下

``` python
>>> pytest test/utils/test_linear_elasticity_sympy.py

test/utils/test_linear_elasticity_sympy.py ........                                                                                                                                                                                                        [100%]
Results (12.47s):
         8 passed

>>> pytest test/utils/test_navier_stokes_sympy.py

test/utils/test_navier_stokes_sympy.py ....                                                                                                                                                                                                                [100%]
Results (17.04s):
         4 passed

>>> python test/utils/speed_test_navier_stokes.py

compute_with_sympy overhead: 0.01575
compute_with_pyfunc overhead: 0.01501
u check pass
v check pass
p check pass
continuity check pass
momentum_x check pass
momentum_y check pass
```

其他：
1. 修复 `LinearElasticity` 内，未正确处理`dim=2`而可能导致KeyError的情况；删除无用的 `navier_x/y/z` 系列表达式，并修改相关单测

@AndPuQing 有空可以review一下

### CODE REVIEW

File: ppsci/utils/sym_to_func.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+        # detach Tensor(s) if specified
+        if self.detach_keys:
+            for key in self.detach_keys:
+                data_dict[key] = data_dict[key].detach()
+
+        return data_dict
+
+
+class ConstantNode(Node):
+    """ "Class for constant variable node in converted expression tree.
```
[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply ppsci/utils/sym_to_func.py]: 已修改

**Review by zhiminzhang0830 — APPROVED** [review]
LGTM
